build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/BuildView.java

```diff
@@ -107,6 +115,21 @@ public String description() {
         return getOrElse(build.getDescription(), "");
     }
 
+    @Override
+    public List<String> stages() {
+        if (this.build instanceof WorkflowRun) {
+            WorkflowRun currentBuild = (WorkflowRun) this.build;
+
+            if (currentBuild.isBuilding()) {
+                StageTraversal traversal = new StageTraversal();
+                traversal.start(currentBuild.getExecution().getCurrentHeads());
```

Comment:
Interesting. Thanks for reporting this!
I haven't seen any NPE on my side. What versions of Jenkins/pipeline plugins are you running? Can you please provide a small sample of your pipeline structure so that I see if I reproduce this?